### PR TITLE
Add annotation for enableCoreDump

### PR DESCRIPTION
**Pull request by ckcd**
See https://github.com/istio/istio/issues/19932

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by rlenglet** [comment]
@ckcd please submit the same PR against the copy of `injection-template.yaml` in https://github.com/istio/installer/blob/master/istio-control/istio-autoinject/files/injection-template.yaml.

**Comment by ckcd** [comment]
> please submit the same PR against the copy of injection-template.yaml in https://github.com/istio/installer/blob/master/istio-control/istio-autoinject/files/injection-template.yaml.

@rlenglet  Sure, will do this once the PR is merged. thanks ~

File: install/kubernetes/helm/istio/files/injection-template.yaml [hunk 1]
```diff
@@ -310,7 +310,7 @@ containers:
       drop:
       - ALL
     privileged: {{ .Values.global.proxy.privileged }}
-    readOnlyRootFilesystem: {{ not .Values.global.proxy.enableCoreDump }}
+    readOnlyRootFilesystem: {{ not (eq (annotation .ObjectMeta `sidecar.istio.io/enableCoreDump` .Values.global.proxy.enableCoreDump) "true") }}
```
Comment (morvencao) [inline install/kubernetes/helm/istio/files/injection-template.yaml]: can we update `not (eq ())` to `ne`?

Comment (ckcd) [reply install/kubernetes/helm/istio/files/injection-template.yaml]: sure, thanks !!

**Comment by ckcd** [comment]
```
// TestController_GetPodLocality/should_return_correct_az_if_node_has_only_subzone_label

controller_test.go:406: wait for pod err: timed out waiting for the condition
```

Don't know why this failed, seems no any relationship with this PR. i will try to retest.

**Pull request merged**